### DISCUSSION TITLE
Freetz auf Fritzbox 6660 Cable mit eigenem Kernel

### DISCUSSION THREAD

**Discussion by dm4tze** [Freetz-NG/freetz-ng#191] [Unerfüllte Wünsche]
Hallo, 
ich würde gerne auf meiner Fritzbox 6660 Cable VLANs verwenden.
Im Wiki steht, man muss hierzu den Kernel ersetzen [Konfiguration des eingebauten Switches](https://freetz.github.io/wiki/help/howtos/security/switch_config.html)

Den Kernel hab ich wie im Wiki beschrieben gebaut: [Kernel konfigurieren und kompilieren](https://freetz.github.io/wiki/help/howtos/development/make_kernel.html)

Damit ich in der menuconfig den eigenen Kernel auswählen konnte, musste ich config/mod/kernel.in anpassen.

Beim Erstellen des Images kommt jetzt allerdings immer folgender Fehler:

`
copying kernel image
  kernel image size: 130.0 MB, max 5.0 MB, free -125.0 MB (-131086126 bytes)
ERROR: kernel image is 131086126 bytes too big. See https://freetz-ng.github.io/freetz-ng/wiki/00_FAQ/FAQ.de.html#filesystem-image-too-big for details.
`
Ist es korrekt der Kernel nur 5 MB groß sein darf. Wie bekomme ich den Kernel so klein? Gibt es hier auch einen Leitfaden oder Konfigurationen, an denen ich mich orientieren kann?

**Comment by fda77** [comment]
Das geht noch gar nicht, da musst du selbst Hand anlegen:

https://freetz-ng.github.io/freetz-ng/CHANGELOG.html#devel
> Known problems:
>    Replace kernel does not work for most latest firmware versions.

Und https://github.com/Freetz-NG/freetz-ng/commit/d3c6920267eecf5764d8a6b4b6b6d0e97ced4819

Aktuell kann man nur eigene Module bauen, der Kernel ist so gross wegen https://github.com/Freetz-NG/freetz-ng/blob/master/make/linux/patches/4.9.199.x86/100-evaloader.TODO.patch

**Reply by fda77** [reply]
PeterPawn schrieb hierzu: https://github.com/PeterPawn/YourFritz/issues/37#issuecomment-776266385
